Use canonical emacs version in JSON recipes

gamb opened the pull request:
Fixes https://github.com/nix-community/emacs-overlay/issues/60

So nix can parse version e.g. for `lib.versionAtLeast` etc.

gamb commented:
Might be some context I'm missing around using "emacs-27.1" as the version name, but fwiw this fixed installs for me.

gamb commented:
Doh. Jumped the gun, I see the `update` script needs to change instead.

gamb commented:
I guess we'd want to fix the other emacs repos. Perhaps this approach is problematic in general

alexvorobiev commented:
The emacs-master's version passes the check:
```
nix-repl> lib.versionAtLeast "20200815.0" "27"
true
```
So as long as that convention is used the patch will be applied.

gamb commented:
@alexvorobiev That's true, in any case I've modified the updater in eace767 so that artifacts resolve as:
```diff
modified   repos/emacs/emacs-feature_native-comp.json
@@ -1 +1 @@
-{"rev": "3882e8fd244a66edb6ba60f40182a4d0772cfcb1", "sha256": "1743xsjzbaclj30b4rd42pnyyi3siv05rcmbkbc3qyghcznrsfcg", "version": "20200814.0"}
+{"rev": "f2e6168ece69d635b4f9d9a138100c6772903d0b", "sha256": "19c536ff4mpjig5vv9qsfmnh68bbqpj1ali5cn220px19h7mkv3j", "version": "28.0.50"}
modified   repos/emacs/emacs-master.json
@@ -1 +1 @@
-{"rev": "5b463af8dda4d66bb63a7ce58177e83a42541bac", "sha256": "009mxmpvss0cj3qbchvfs9hds8yyi5p5c6xhjghxp39y0zf6ldk7", "version": "20200815.0"}
+{"rev": "748afc183c2c44b7b2a582d3078cf3d8b4d5270a", "sha256": "0ja7k9hza93lay7x7w99s18wzn4bqrka548f9ms8llld9rs99ass", "version": "28.0.50"}
modified   repos/emacs/emacs-unstable.json
@@ -1 +1 @@
-{"rev": "emacs-27.1", "sha256": "1i50ksf96fxa3ymdb1irpc82vi67861sr4xlcmh9f64qw9imm3ks", "version": "27.1"}
+{"rev": "86d8d76aa36037184db0b2897c434cdaab1a9ae8", "sha256": "1i50ksf96fxa3ymdb1irpc82vi67861sr4xlcmh9f64qw9imm3ks", "version": "27.1"}
```

gamb commented:
Perhaps we could also check this in `test.nix` with something like:
```diff
 { pkgs ? import <nixpkgs> { overlays = [ (import ../../default.nix) ]; } }:
 
 let
+  lib = pkgs.stdenv.lib;
+
   mkTestBuild = package: let
     emacsPackages = pkgs.emacsPackagesNgGen package;
     emacsWithPackages = emacsPackages.emacsWithPackages;
-  in emacsWithPackages(epkgs: [ ]);
+  in assert (lib.versionAtLeast package.version "27"); emacsWithPackages(epkgs: [ ]);
 
 in {
   emacsUnstable = mkTestBuild pkgs.emacsUnstable;
```

etu commented on repos/emacs/update:
```diff
@@ -11,29 +11,27 @@ function update_repo() {
+    { read -r digest; read -r buildPath; } < <(nix-prefetch-url --print-path --unpack "https://github.com/$owner/$repo/archive/${commit_sha}.tar.gz")
+
+    # Grep for Emacs version inside README
+    version=$(cat ${buildPath}/README | grep -Po "This directory tree holds version \K[\d\.]+")
```
I'm thinking that something like this would be more solid and less subject to change than a README.

```suggestion
    version=$(grep -Po "GNU Emacs, \K[\d\.]+" ${buildPath}/configure.ac)
```

gamb replied:
Thanks, couldn't see a solid candidate but configure.ac makes sense :+1:

gamb commented:
cc @adisbladis happy to merge this?

adisbladis commented:
@gamb Sorry this took me a while to get to looking at. Could you rebase it and fix the conflicts? I think this PR looks good to go otherwise.

gamb commented:
Hey @adisbladis no worries! You know I think this has been magically fixed already, in: https://github.com/nix-community/emacs-overlay/commit/8298ba46b685f1fcedccb8934b52a3dbd36b7fc4
